float32 dp
int32 minDistx
int32 minDisty
int32 param1
int32 param2
int32 minRadius
int32 maxRadius
